Assets/Scripts/Projectile.cs

```diff
@@ -0,0 +1,91 @@
+using UnityEngine;
+
+[RequireComponent(typeof(Hit))]
+[RequireComponent(typeof(Rigidbody2D))]
+public class Projectile : MonoBehaviour
```

Comment:
I realise that this is a first pass at projectiles for testing purposes, but is it worth having a projectile pool and pulling from that? Doesn't have to be implemented in this PR